hal/mbed_usb_phy.cpp

```diff
@@ -0,0 +1,28 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "usb_phy_api.h"
+#include "mbed_error.h"
+
+#if !defined(DEVICE_USBDEVICE) || !DEVICE_USBDEVICE
+
```

Comment:
Where is this macro originally defined?

Reply:
Targets which support USB device mode add `USBDEVICE` to `device_has` in targets.json. The build system then creates the define `DEVICE_USBDEVICE`.